server/ladder_service.py

```diff
@@ -131,60 +138,73 @@ def _cancel_existing_searches(self, initiator: Player) -> List[Search]:
+        async for s1, s2 in self.queues[queue_name].iter_matches():
             try:
-                assert len(s1.players) == 1
-                assert len(s2.players) == 1
-                p1, p2 = s1.players[0], s2.players[0]
-                msg = {"command": "match_found", "queue": "ladder1v1"}
+                msg = {"command": "match_found", "queue": queue_name}
                 # TODO: Handle disconnection with a client supported message
-                await asyncio.gather(
-                    p1.send_message(msg),
-                    p2.send_message(msg)
+                await asyncio.gather(*[
+                    player.send_message(msg)
+                    for player in s1.players + s2.players
+                ])
+                asyncio.ensure_future(
+                    self.start_game(s1.players, s2.players, queue_name)
                 )
-                asyncio.ensure_future(self.start_game(p1, p2))
             except Exception as e:
                 self._logger.exception(
                     "Error processing match between searches %s, and %s: %s",
                     s1, s2, e
                 )
 
-    async def start_game(self, host: Player, guest: Player):
+    async def start_game(self, team1: List[Player], team2: List[Player], queue: str):
+        self._logger.debug(
+            "Starting %s game between %s and %s", queue, team1, team2
+        )
         try:
-            self._logger.debug(
-                "Starting ladder game between %s and %s", host, guest
-            )
+            host = team1[0]
+            all_players = team1 + team2
+            all_guests = all_players[1:]
+
             host.state = PlayerState.HOSTING
-            guest.state = PlayerState.JOINING
+            for guest in all_guests:
+                guest.state = PlayerState.JOINING
 
-            (map_id, map_name, map_path) = await self.choose_map([host, guest])
+            # TODO: Need a different map pool for team matchmaker?
+            (_, _, map_path) = await self.choose_map(all_players)
 
+            # TODO: Different game mode for team matchmaker?
             game = self.game_service.create_game(
-                game_mode='ladder1v1',
+                game_mode=queue,
                 host=host,
-                name=f"{host.login} Vs {guest.login}"
+                name=self.game_name(team1, team2)
             )
+            game.init_mode = InitMode.AUTO_LOBBY
+            game.map_file_path = map_path
 
-            host.game = game
-            guest.game = game
+            for i, player in enumerate(all_players):
+                i += 1  # Game options are 1-indexed
+                player.game = game
 
-            game.map_file_path = map_path
+                # Configure game options
+                game.set_player_option(player.id, 'Faction', player.faction)
+                game.set_player_option(player.id, 'Color', i)
+                game.set_player_option(player.id, 'Army', i+1)
```

Comment:
I think it actually should be just `i`. I put `i+1` because that's what the old ladder 1v1 code was doing. However, It turns out that the autolobby code would actually change that configuration back so that the ArmyID was the same as the StartSpot.